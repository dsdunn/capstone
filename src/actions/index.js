export const signIn = (user) => ({
  type: 'SIGN_IN',
  user
})

export const signOut = () => ({
  type: 'SIGN_OUT'
})

export const updateUser = (user) => ({
  type: 'UPDATE_USER',
  user
})

export const setProfile = profile => ({
  type: 'SET_PROFILE',
  profile
})

export const addCollections = (collections) => ({
  type: 'ADD_COLLECTIONS',
  collections
})

export const setCollection = collection => ({
  type: 'SET_COLLECTION',
  collection
})

export const updateCollectionsList = (collectionsList) => {
  type: 'UPDATE_COLLECTIONS_LIST',
  collectionsList
}