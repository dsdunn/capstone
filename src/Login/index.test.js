import React from 'react';
import { shallow } from 'enzyme';
import { signIn, setProfile } from '../actions';
import { Login, mapStateToProps, mapDispatchToProps } from './';

describe('Login', () => {
  let wrapper;
  let mockEvent;
  let history;

  beforeEach(() => {
    history = { location: { pathname: '/'},
                goBack: jest.fn(),
                push: jest.fn() 
              }


    mockEvent = { preventDefault: jest.fn(),
                  target: {
                    id: 'email',
                    value: 'me@you.com'
                  }
                }

    wrapper = shallow(<Login history={history} />)
  })

  it('should start with default state of email, password, error', () => {
    const expected = {
      email: '',
      password: '',
      error: ''
    }
    const actual = wrapper.state()

    expect(actual).toEqual(expected)
  })

  it('should update state when handleChange is called', () => {
    wrapper.instance().handleChange(mockEvent)
    const actual = wrapper.state()
    const expected = { email: 'me@you.com', password: '', error: '' }
    expect(actual).toEqual(expected)
  })

  it('should reset the form when resetForm is called', () => {
    wrapper.setState({ email: 'me@you.com' })
    wrapper.instance().resetForm()
    const actual = wrapper.state('email')
    const expected = ''

    expect(actual).toEqual(expected)
  })

  it('should return a props object with signIn', () => {
    const mockDispatch = jest.fn();
    const mockUser = { password1: '123456',
                       password2: '123456',
                       email: 'heck@yeah.com',
                       username: 'Sparkle' }

    const mappedProps = mapDispatchToProps(mockDispatch)
    const actionToDispatch = signIn(mockUser);

    mappedProps.signIn(mockUser);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })

  it('should return a props object with setProfile', () => {
    const mockDispatch = jest.fn();
    const mockUser = { password1: '123456',
                       password2: '123456',
                       email: 'heck@yeah.com',
                       username: 'Sparkle' }
                       

    const mappedProps = mapDispatchToProps(mockDispatch)
    const actionToDispatch = setProfile(mockUser);

    mappedProps.setProfile(mockUser);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })

  it('should call goBack when the form is closed', () => {
    let spy = jest.spyOn(wrapper.instance(), 'goBack')
    wrapper.find('.login-close-link').simulate('click');

    expect(spy).toHaveBeenCalled();
  })

  it('should go call goBack when goBack', () => {
    wrapper.instance().goBack()
    expect(history.goBack).toHaveBeenCalled()
  })

  it('should match the snapShot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should return a props object', () => {
    const mockState = {user: {}};
    const expected = {"user": {}};

    const mappedProps = mapStateToProps(mockState)

    expect(mappedProps).toEqual(expected)
  })
})
