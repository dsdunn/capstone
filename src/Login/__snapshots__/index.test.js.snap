// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapShot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    history={
        Object {
            "goBack": [Function],
            "location": Object {
              "pathname": "/",
            },
            "push": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <a
          className="login-close-link"
          exact="true"
          onClick={[Function]}
>
          <img
                    className="close-btn"
                    src="close.svg"
          />
</a>,
        <h5>
          Login
</h5>,
        <label
          for="email"
>
          email
</label>,
        <input
          id="email"
          onChange={[Function]}
          value=""
/>,
        <label
          for="password"
>
          password
</label>,
        <input
          id="password"
          onChange={[Function]}
          type="password"
          value=""
/>,
        <button>
          Submit
</button>,
        <p
          className="login-err-message"
/>,
      ],
      "className": "login",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            className="close-btn"
            src="close.svg"
/>,
          "className": "login-close-link",
          "exact": "true",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "close-btn",
            "src": "close.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login",
        },
        "ref": null,
        "rendered": "Login",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "email",
          "for": "email",
        },
        "ref": null,
        "rendered": "email",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "email",
          "onChange": [Function],
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "password",
          "for": "password",
        },
        "ref": null,
        "rendered": "password",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "password",
          "onChange": [Function],
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit",
        },
        "ref": null,
        "rendered": "Submit",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "login-err-message",
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            className="login-close-link"
            exact="true"
            onClick={[Function]}
>
            <img
                        className="close-btn"
                        src="close.svg"
            />
</a>,
          <h5>
            Login
</h5>,
          <label
            for="email"
>
            email
</label>,
          <input
            id="email"
            onChange={[Function]}
            value=""
/>,
          <label
            for="password"
>
            password
</label>,
          <input
            id="password"
            onChange={[Function]}
            type="password"
            value=""
/>,
          <button>
            Submit
</button>,
          <p
            className="login-err-message"
/>,
        ],
        "className": "login",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              className="close-btn"
              src="close.svg"
/>,
            "className": "login-close-link",
            "exact": "true",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "close-btn",
              "src": "close.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
          },
          "ref": null,
          "rendered": "Login",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "email",
            "for": "email",
          },
          "ref": null,
          "rendered": "email",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "email",
            "onChange": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "password",
            "for": "password",
          },
          "ref": null,
          "rendered": "password",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "password",
            "onChange": [Function],
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "login-err-message",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
